lib/core/sdam/server_description.js

```diff
@@ -180,7 +180,7 @@ function parseServerType(ismaster) {
   if (ismaster.setName) {
     if (ismaster.hidden) {
       return ServerType.RSOther;
-    } else if (ismaster.ismaster) {
+    } else if (ismaster.ismaster || ismaster.isWritablePrimary) {
```

Comment:
does this fix a bug or implement a feature?

Reply:
I'll be honest this line was lifted from the other helloOk PR, I'll investigate, since we always patch in `ismaster.ismaster = ismaster.isWritablePrimary` whenever the hello comes in from the server my bet is that this was added to satify tests that bypass that section, but let me confirm and circle back

Reply:
That's exactly what this was for, SDAM tests simulate the Topology directly so the patch is never made, I would move it to the testing files but I feel like such a small safety net should be kept in driver. However I think it should be patched in the ServerDescription constructor instead of an `||` check. I can open a ticket so we make sure 4.x is sync-ed if you think that change is worth it.